### ISSUE TITLE
iPad: Load first story when opening feed

### ISSUE THREAD

**Issue by samuelclay** [samuelclay/NewsBlur#1500]
Since the story content view is visible when a feed/folder is selected, let's just load the top story when it comes in. As soon as the user scrolls and touches the view, the story is marked as read. If they don't touch the view but switch to another story, it's not marked as read. Hitting the Next Unread button similarly doesn't mark it as read. Tapping the story title does mark it as read, though.

**Comment by Dejal** [comment]
I'm out of time, but there's a work-in-progress on this. It selects the first story, suppressing marking as read, but currently doesn't mark as read on feed detail selection or story scroll.

Let me know what you think after trying this; while showing the first story is a nice improvement, I'm not sure that not marking it as read is the right approach, since a short story could be read without any interaction. The requested behavior introduces a bunch of unusual states.

**Comment by samuelclay** [comment]
I think I'm OK with a short story not being marked as read if the user taps on another story title. If they want to mark it as read, they just need to tap anywhere in the story itself. That mirrors the functionality on the web.

**Comment by nriley** [comment]
I'd like it if when you navigate to another story with the keyboard it marks the story as read. Otherwise I end up with the first story "accidentally" unread.

**Comment by samuelclay** [comment]
That's the point of this feature though, is that it leaves the first story unread unless you tap inside it or scroll it.

**Comment by nriley** [comment]
This means I need to mark the first story read in a completely different way (from the keyboard, pressing 'm') than every other story. It seems inconsistent.

**Comment by nriley** [comment]
Aha! Totally missed that it was an option on the web! I do indeed have "Show all stories" selected…

**Comment by samuelclay** [comment]
So is the "tap-to-mark-read" feature not enough? You can hit 'u' or 'm' to mark it as read, or even 'space' to mark it as read and move to the next story.

**Comment by samuelclay** [comment]
I agree this is confusing but the alternative is to not show the first story to force you to open it, or to mark it as read, which should be a preference.

**Comment by nriley** [comment]
Thanks, I don't use space except to scroll and hadn't tried. I guess I can get used to it. Given the choice I'd probably just not show the first story to force me to open it, but I realize I'm in the minority in reading primarily with an iPad with attached keyboard.

**Comment by samuelclay** [comment]
I think it's worthwhile to cater to you because otherwise I feel like it's unfinished. Plus we just added this. So let's add a preference and make it like the web. The first option doesn't show the story, the second option does show the story but also marks it as read. We could also have a third middle option which is today's feature, which is to show the story but not mark as read.

@Dejal, is it trivial to implement all three based on the changes made to get us here? Do you think it's too much? Seems close to the line of too much, but as a reading app, it should cater to many different needs. I mean just look at how people reacted to font and spacing changes. I had to add two new controls after I launched the beta site (comfortable/compact and feed title font).

**Comment by Dejal** [comment]
Adding a preference often feels like indecisiveness to me, but I agree that people can have strong opinions on how they want it to work.

"Open the first story" sounds like it shows the story, and "show all stories" sounds like it doesn't. Or am I misinterpreting?

But yes, it wouldn't be too hard to add three options, to not select the first story (as it was before), to select the first story but not mark as read (as now), or select the first story and mark as read.

Would you want that in the Settings, in the feed detail menu, or both?

**Comment by samuelclay** [comment]
Let's make it two options: Show all stories / Open the first story. Let's default to Open the first story on iPad and Show all stories on iPhone. So that means dropping the logic you wrote for tapping to mark as read and only using the two extremes.

**Comment by nriley** [comment]
Yes, that's how it behaves on the Web. My preference would be the existing "show all stories" behavior so I explicitly have to display the first story. Open and mark read would be less preferred, with the current behavior (open and don't mark read) is least preferred. Currently the Web site will mark as read when opening the first story, and iPad doesn't. The current iPad behavior matches neither of the Web site options— it'll open and NOT mark read.

Nicholas

**Comment by samuelclay** [comment]
We should have an option on iPhone (and iPad) to automatically open All Site Stories, as well as an option to automatically open the first unread story.

Also, when I opened the latest build on iPad, the preference was automatically set to "Open first story" but it didn't open. Then when I switched between preferences and back to open first story, it worked. Shouldn't there be a default?

**Comment by samuelclay** [comment]
> We should have an option on iPhone (and iPad) to automatically open All Site Stories, as well as an option to automatically open the first unread story.

Just to clarify, those two preferences on both ipad and iphone, which means we won't auto-open all site stories on ipad if somebody turns it off. So open All Site Stories is on by default on ipad, and off by default on iphone.

**Comment by Dejal** [comment]
I'm not sure I understand. The options that were requested before are when a user taps a feed or folder, it either shows the list of stories (feed detail), or automatically shows the first story. Both of which are implemented and work well on iPad. On iPhone, showing the first story instead of the list of stories is trickier, but I can make it work if desired.

Now you seem to be talking about a new option, to show the All Stories river on launch, instead of showing the feeds list? Did you mean to request something new, or did you conflate things?

**Comment by Dejal** [comment]
Furthermore, I would recommend that we don't add any more new features to NewsBlur for this release. Let's lock down the current features, rather than adding risky new ones. So I think we should leave this as-is, and come back to further tweaks, like adding the preference on iPhone, after version 11.1 is out.

**Comment by samuelclay** [comment]
I think we have only today left to work on this release but I'd like to push on this change for the "auto-load first unread story preference". I realized once I used it regularly that showing the story means I'm going to read it without tapping and then use whatever mechanism I normally use to read the next story, leaving the original story unread. We shouldn't show anything with that preference as it is on the web. But the default, both on the web and on iOS, is to auto-open the first unread story. 

About the other preference for "automatically open folder", that should be a new ticket, which I've created here: #1604
